### ISSUE TITLE
Incompatibilities with search and Sphinx 5.x

### ISSUE THREAD

**Issue by stsewd** [readthedocs/readthedocs.org#8871]
TL;DR: Sphinx is dropping jQuery https://github.com/sphinx-doc/sphinx/pull/10028/ and those changes break our custom server side search feature (SSS), and it also breaks the search page on our theme (with or without our SSS).

The first problems were reported at https://github.com/readthedocs/readthedocs.org/issues/8862.

- `Seach.init` is being called twice, duplicating some elements (will be fixed with https://github.com/readthedocs/readthedocs-sphinx-ext/pull/100)
- Our custom js that replaces Sphinx's default search with our SSS assumes that some attributes from the `Search` object are jQuery objects, but after the refactor done upstream, they will be HTML/Node object

  We need to check all usages of `Sphinx.*` and decide what to do. 

  - https://github.com/readthedocs/readthedocs.org/blob/e7102a4d7f0c5169f6b54796a43b39f7f88d4c70/readthedocs/core/static-src/core/js/doc-embed/search.js#L209
  All calls to `Sphinx.{element}.text()` should be easy to fix, we can have a utility function that checks if the element is an jQuery object, something like

      ```js
      const setText = function(node, text) {
        if (node.jquery) {
          node.text(text);
        } else {
          node.innerText = text;
        }
      };
      ```

  - https://github.com/readthedocs/readthedocs.org/blob/e7102a4d7f0c5169f6b54796a43b39f7f88d4c70/readthedocs/core/static-src/core/js/doc-embed/search.js#L209-L211

    I think we can delete that call to `.fadeIn`, it's done at the end of the search operation, so at this stage the text from `Sphinx.status` is already shown. Actually, I think this was meant to be `Sphinx.title.fadeIn`.

  -  https://github.com/readthedocs/readthedocs.org/blob/e7102a4d7f0c5169f6b54796a43b39f7f88d4c70/readthedocs/core/static-src/core/js/doc-embed/search.js#L191-L192
    
     `list_item` is a jQuery object (we are making heavy use of jQuery to create that element), but now `Search.output` will be an HTML/Node object, we can do something like this:
        
        ```js
        if (Search.output.jquery) {
          Search.output.append(list_item);
          list_item.slideDown(5);
        } else {
          Search.output.appendChild(list_item.get(0));
          // No slideDown for the non-jquery version.
        }
        ```
- If some expected elements aren't on the search page, the search operation will fail.
  
    https://github.com/sphinx-doc/sphinx/blob/3b01fbe2adf3077cad5c2cf345c6d000d429d7ac/sphinx/themes/basic/static/searchtools.js#L211-L213
  
  Our theme completely overrides the search page, and it doesn't inherit from Sphinx's default search page

   https://github.com/readthedocs/sphinx_rtd_theme/blob/d64dadf1ceec4f9ff6c1ca2d3ea4c3d0fdb0e8d2/sphinx_rtd_theme/search.html#L10

   When retrieving a non-existent element, jQuery would still return an object (kind of empty one, so method calls won't raise a null exception), but now `getElementById` will return  `null` and raise an exception when trying to call a method on that value.

   We can fix this in our theme (by adding that element or by inheriting from sphinx's search page to avoid things like this in the future),
   or we can also send a patch to sphinx to not assume that that element will always be present (it's used in 2 places only, so it shouldn't be a problem).

Those are the issues I have found so far, additionally Sphinx will soon be dropping jQuery too (and we make heavy use of jQuery in our injected footer and friends).

**Comment by pradyunsg** [comment]
FWIW, this also affects Sphinx 4 sites.

**Comment by stsewd** [comment]
@pradyunsg do you have a link to those projects? What are the problems present on 4.x?
